File: inventario_frontend/src/models/purchase.ts
import { Product } from "./product";
import { Client } from "./client";

export class PurchaseForCreation {
  public productId: string | undefined = undefined;
  public product: Product | undefined = undefined;
  public clientId: string | undefined = undefined;
  public client: Client | undefined = undefined;
  public clientName: string | undefined = undefined;
  public date: Date | undefined = undefined;
  public quantity: number | undefined = undefined;
  public amount: number | undefined = undefined;
}

export class Purchase extends PurchaseForCreation {
  public id!: string;
}

File: Inventario/DataAccess/Models/Purchase.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Data
{
    public class Purchase: Entity
    {
        public Guid ProductId { get; set; }
        public Product Product { get; set; }
        public Guid? ClientId { get; set; }
        public Client Client { get; set; }
        public string ClientName { get; set; }
        public DateTime Date { get; set; }
        public double Quantity { get; set; }
        public double Amount { get; set; }
    }
}
